pkg/engine/pattern.go

```diff
@@ -91,6 +92,14 @@ func validateValueWithIntPattern(value interface{}, pattern int64) bool {
 
 		glog.Warningf("Expected int, found float: %f\n", typedValue)
 		return false
+	case string:
+		// extract int64 from string
+		int64Num, err := strconv.ParseInt(typedValue, 10, 64)
```

Comment:
If I set memory limit in policy to "1Gi", and the resource has the limit set to "1024Mi", would this work? Same for the cpu "200m" vs "0.2" ?

Reply:
@realshuting 
- first, value `1Gi` will not be compared in validateValueWithIntPattern as its an alphanumeric and will be a string comparison. This check if for numbers types only.
- second, here we parsing through each node and comparing the values based on types( custom parser, which is independent of the value string contains). The logic to compare Gi and Mi is kubernetes specific and will be handheld by the server. So if we cannot to Gi and Mi comparison here. If it is required, u'll need to create specifically handle these scenarios.

Reply:
Ok, if this pr is to fix numbers only, we should be good.

> The logic to compare Gi and Mi is kubernetes specific and will be handheld by the server.

As Kyverno is doing the validation checks in webhook, it seems like we have to handle this before k8s, created an issue to tack #428.